src/components/SingleTopic.jsx

```diff
@@ -9,16 +9,25 @@ const SingleTopic = () => {
     const [loading, setLoading] = useState(true);
     const [sort, setSort] = useState('date');
     const [sortOrder, setSortOrder] = useState('desc');
+    const [error, setError] = useState();
 
     useEffect(() => {
         getArticlesByTopic(topic, sort, sortOrder).then((articlesByTopicFilter) => {
-            setTopicFilter(articlesByTopicFilter.articles);
-            setLoading(false);
+            if(articlesByTopicFilter.articles){
+                setTopicFilter(articlesByTopicFilter.articles);
+                setLoading(false);
```

Comment:
You need to setError to false here as well or you can get caught in an error page